### DISCUSSION TITLE
Seems like the migrator addColumn() is not working

### DISCUSSION THREAD

**Discussion by madsane29** [simolus3/drift#2058] [Q&A]
Hello! In my app (version 1.0.0) I had 2 tables, then I modified these tables (app version 2.0.0) by adding 1 extra column to TableA and 2 extra columns to TableB.

I wrote the `await m.addColumn(table, table.column);` with increased schema version but if I update the 1.0.0 version to 2.0.0, the extra columns are not created.

Based on https://drift.simonbinder.eu/docs/advanced-features/migrations/#basics  I'd not have to do more but I tried to `m.alterTable(...)` these tables... Maybe... Still no result.

I wonder what could go wrong?

![image](https://user-images.githubusercontent.com/40585482/190233067-565d7858-25e4-401b-9a25-475fb78570e8.png)

![image](https://user-images.githubusercontent.com/40585482/190233329-fcf625ec-ef36-454d-8cfd-b5cd085007ca.png)
![image](https://user-images.githubusercontent.com/40585482/190233400-741a457d-4030-4470-b815-a6e7b7202d7b.png)

Thanks in advance.

**Comment by simolus3** [comment]
Did you set a breakpoint in the migration to check that it's actually running? Do any errors get reported during the migration?

It's hard to say for sure what's going wrong here. Nothing obvious stands out from your code. Did you increase the schema version and run the app before writing the migration? In that case the database file will already have the new version without the migration ever running. You can run a custom select with `pragma user_version` to see which schema version is installed on the database.

**Reply by madsane29** [reply]
This problem was found when a tester installed then new version on his phone and some strange behavior the app showed.

So I believe this part could not cause the problem: `Did you increase the schema version and run the app before writing the migration?`

Also after I heard about this problem I installed the version 1.0.0, then updated it to 2.0.0 with the IPA installer and also did not create the extra columns.

Later / tomorrow I will check that pragma thing. Should I just do it like this: `db.customStatement('pragma user_version');` ?

Also I will check if migrator throws any errors. Thanks.

**Reply by madsane29** [reply]
Found the problem... An exception was thrown during the migration (`onUpgrade` part) so the `addColumn()` lines did not even run. Thanks for the help. So no problem, it was my bad. Thanks.

**Reply by simolus3** [reply]
> An exception was thrown during the migration

Drift should have rethrown that exception so that it gets reported by the first call to the database responsible for opening it. If you didn't see that exception initially, that might also be a problem in in drift (unless you're ignoring that exception somehow)?

**Reply by madsane29** [reply]
Rethrown where?

I did not test the app by updating the older version with the newer one...

**Reply by simolus3** [reply]
Drift lazily opens the database when it's first used (so at the point where the first SQL query is made). If the migration throws an exception, that query should complete with that exception.

What I'm saying is that drift shouldn't hide exceptions happening in the migration from you - that would be terrible. If that's what has happened here, please let me know so that I can take a look and fix it :)

**Reply by madsane29** [reply]
Nah, it works as it should. After I installed the old version with APK, then installed the new version through android studio, I saw the exception.
